src/jackdaw/specs.clj

```diff
@@ -31,11 +32,20 @@
 
 ;; Avro serde
 
-(s/def :jackdaw.serde/confluent-avro-serde
-  (s/keys :req-un [::serde-keyword
-                   ::key?
-                   (or ::schema ::schema-filename)]))
+(defn exactly-one-true?
+  [& args]
+  (= 1 (count (filter identity args))))
 
+(s/def :jackdaw.serde/confluent-avro-serde
+  (s/and
+   (s/keys :req-un [::serde-keyword
+                    ::key?
+                    (or ::schema
+                        ::schema-filename
+                        ::read-only?)])
+   #(exactly-one-true? (string? (:schema %))
+                       (string? (:schema-filename %))
+                       (true? (:read-only? %)))))
```

Comment:
I thought we agreed just to check for the key's presence or absence.

Reply:
I misunderstood - easy enough to fix